sprocs/instance_questions_points_exam.sql

```diff
@@ -23,31 +23,40 @@ BEGIN
     SELECT * INTO aq FROM assessment_questions WHERE id = iq.assessment_question_id;
 
     max_points := COALESCE(aq.max_points, 0);
-    highest_submission_score := greatest(submission_score, coalesce(highest_submission_score, 0));
 
-    correct := (submission_score >= 0.5);
+    -- Update points (instance_question will be closed when number_attempts exceeds bound,
+    -- so we don't have to worry about accessing a non-existent entry in points_list_original,
+    -- but use coalesce just to be safe)
+    points := coalesce(iq.points, 0) + (coalesce(iq.points_list_original[coalesce(iq.number_attempts, 0)+1], 0) * GREATEST(0, submission_score - coalesce(iq.highest_submission_score, 0)));
```

Comment:
I think we should add a special case after this points computation like:
```sql
IF (submission_score >= 1) AND (iq.points_list_original[iq.number_attempts + 1] = max_points) THEN
    points := max_points;
END IF
```

This is for the case when `points_list = [10, 10, 10, 10]` and the student gets scores `20%, 40%, 80%, 100%`. We want this to give exactly 10 points, but we could have floating point issues from computing it incrementally.

Reply:
I understand. (Note - I don't think check anyway that `points_list` is non-increasing. If it isn't, results are going to be strange, here and elsewhere.)

Reply:
I just filed #773 for this.